DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Rules/AnonymizerRuleFactory.cs

```diff
@@ -0,0 +1,136 @@
+            _processorFactory = processorFactory;
+            _settingsFactory = settingsFactory;
+        }
+
+        public AnonymizerRule[] CreateAnonymizationDicomRule(JObject[] ruleContent)
+        {
+            return ruleContent?.Select(entry => CreateAnonymizationDicomRule(entry)).ToArray();
+        }
+
+        public AnonymizerRule CreateAnonymizationDicomRule(JObject ruleContent)
+        {
+            EnsureArg.IsNotNull(ruleContent, nameof(ruleContent));
+
+            // Parse and validate method
+            if (!ruleContent.ContainsKey(Constants.MethodKey))
+            {
+                throw new AnonymizationConfigurationException(DicomAnonymizationErrorCode.MissingConfigurationFields, "Missing method in rule config");
+            }
+
+            var method = ruleContent[Constants.MethodKey].ToString();
+            var supportedMethods = Enum.GetNames(typeof(AnonymizerMethod)).ToHashSet(StringComparer.InvariantCultureIgnoreCase);
+            if (!supportedMethods.Contains(method))
+            {
+                throw new AnonymizationConfigurationException(DicomAnonymizationErrorCode.UnsupportedAnonymizationRule, $"Anonymization method {method} not supported.");
+            }
+
+            // Parse and validate settings
+            JObject parameters = null;
+            if (ruleContent.ContainsKey(Constants.Parameters))
+            {
+                parameters = ruleContent[Constants.Parameters].ToObject<JObject>();
+            }
+
+            JObject ruleSetting = _defaultSettings.GetDefaultSetting(method);
+            if (ruleContent.ContainsKey(Constants.RuleSetting))
+            {
+                if (_customizedSettings == null || !_customizedSettings.ContainsKey(ruleContent[Constants.RuleSetting].ToString()))
+                {
+                    throw new AnonymizationConfigurationException(DicomAnonymizationErrorCode.MissingRuleSettings, $"Customized setting {ruleContent[Constants.RuleSetting]} not defined");
+                }
+
+                ruleSetting = _customizedSettings[ruleContent[Constants.RuleSetting].ToString()];
+            }
+
+            if (parameters != null)
+            {
+                if (ruleSetting == null)
+                {
+                    ruleSetting = parameters;
+                }
+                else
+                {
+                    ruleSetting.Merge(parameters, new JsonMergeSettings { MergeArrayHandling = MergeArrayHandling.Concat });
+                }
+            }
+
+            // Parse and validate tag
+            if (ruleContent.ContainsKey(Constants.TagKey))
+            {
+                var content = ruleContent[Constants.TagKey].ToString();
+
+                try
+                {
+                    var tag = DicomTag.Parse(content);
+                    return new AnonymizerTagRule(tag, method, ruleSetting, ruleContent.ToString(), _processorFactory, _settingsFactory);
+                }
+                catch (Exception)
```

Comment:
you don't need to do this nested try catch, seems like you can simply seperate them in different blocks.

Reply:
agreed. Deep nested blocks make the code difficult to read and maintain.

Reply:
Added tryParse functions to avoid nested try catch.